### PR TITLE
Replace Angular code that synchronizes URL parameters with Time API

### PR DESCRIPTION
**Pull request by akhenry**
Fixes https://github.com/nasa/openmct/issues/3088

This [started as a PR to add a simple test to telemetry tables](https://github.com/nasa/openmct/pull/2989), but that test was blocked by a number of issues that are addressed in this PR.  These issues collectively were causing what appeared to be race conditions in automated tests that were dependent upon time API state (eg. any tests that involved changing time bounds). In affected tests, the bounds would be set in test setup but seem to randomly revert to default bounds (ie. `{start: 0, end: 1}`) between test setup and when the tests were actually run.

* The existing code that kept the Time API and URL in sync relied on Angular digests, which are unpredictable. This code has been replaced with code that does not rely upon Angular JS.
* The URL was causing state to leak between tests, even when the Open MCT instance was being cleaned up. This has been addressed by adding a new `resetApplicationState` function that should be called after each test.

This adds some new utility functions / classes:
* The [`resetApplicationState`](https://github.com/nasa/openmct/blob/b26f8eb9465eb450e9f5868e562ae57e673bdadc/src/utils/testing.js#L59) function will reset the URL, and clean up application-wide listeners. It calls a new `destroy` function on Open MCT.
* [`openmctLocation`](https://github.com/nasa/openmct/blob/new-table-test/src/utils/openmctLocation.js) provides functions for getting and setting the navigated object path, as well as individual search parameters.

Finally, table test cases are updated with a new test that checks that telemetry data is correctly added as rows to the table.

### Author Checklist
1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y
5. Testing instructions included?

### CODE REVIEW

File: src/plugins/condition/pluginSpec.js [hunk 1]
```diff
@@ -20,19 +20,23 @@
...
 
-import { createOpenMct } from "testUtils";
+import { createOpenMct, resetApplicationState } from "utils/testing";
 import ConditionPlugin from "./plugin";
 import StylesView from "./components/inspector/StylesView.vue";
 import Vue from 'vue';
 import {getApplicableStylesForItem} from "./utils/styleUtils";
 
-describe('the plugin', function () {
+xdescribe('the plugin', function () {
```
Comment (deeptailor) [inline src/plugins/condition/pluginSpec.js:29]: Was the x - left by mistake?

File: src/plugins/notificationIndicator/pluginSpec.js [hunk 2]
```diff
@@ -23,17 +23,22 @@
...
 import Vue from 'vue';
 import {
-    createOpenMct
-} from 'testUtils';
+    createOpenMct,
+    resetApplicationState
+} from 'utils/testing';
 
-describe('the plugin', () => {
+xdescribe('the plugin', () => {
```
Comment (deeptailor) [inline src/plugins/notificationIndicator/pluginSpec.js:30]: see comment above

File: src/plugins/condition/pluginSpec.js (hunk 1 above)
Comment (akhenry) [reply src/plugins/condition/pluginSpec.js]: Good catch! Thanks

**Comment by deeptailor** [comment]
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? Yes
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes
Has associated issue been labelled unverified? (only applicable if this PR closes the issue) No

**Comment by deeptailor** [comment]
@akhenry - Missing some testing notes

**Pull request merged**